### PR TITLE
Change Bitcoin identity to a Public Key

### LINKED ISSUE #74: Bitcoin "identity" should be a SECP256k1 public key
Right now our we use the "pubkey hash" as the identity in [RFC005](https://github.com/comit-network/RFCs/blob/master/RFC-005-SWAP-Basic-Bitcoin.md). This is the wrong choice for two reasons:

1. In P2SH scripts the public key is already "hashed". We should just lock the branches of the if statement to the public keys themselves. (miniscript for example also generates scripts that use public keys because the witness stacks are smaller and hence cheaper)
2. In the [taproot proposal](https://github.com/sipa/bips/blob/bip-schnorr/bip-taproot.mediawiki) outputs are locked to public keys. So sending to someone's public key will become the standard way of transferring Bitcoin (hopefully).

Related to: https://github.com/comit-network/comit-rs/issues/1296

## Recommendation

This is just about updating the RFC.
Note that we still use the hash in the HTLC, however the `Identity` is now a public key.
Also check the HTLC section as we may need to add a note that we don't use the `Identity` but the hash of it.

### PR DESCRIPTION
**Pull request by D4nte**
I changed my mind 3 times on whether to use `Public Key`, `pubkey` or `public key`. Let's see what the team thinks :)

Resolves #74

### CODE REVIEW

File: RFC-005-SWAP-Basic-Bitcoin.adoc [hunk 1]
```diff
@@ -22,13 +22,12 @@ To fulfil the requirements of link:./RFC-003-SWAP-Basic.md[RFC003] this RFC defi
 
 link:./RFC-003-SWAP-Basic.md[RFC003] requires ledgers have an *identity* type specified to negotiate a SWAP.
 
-The Identity to be used on Bitcoin is the 20-byte *pubkeyhash* which is the result of applying SHA-256 and then RIPEMD-160 to a user's compressed SECP256k1 public key.
+The Identity to be used on Bitcoin is the user's compressed SECP256k1 Public Key, or `pubkey`.
 The compressed public key is used because it needs to be compatible with segwit transactions (see https://github.com/bitcoin/bips/blob/master/bip-0143.mediawiki#Restrictions_on_public_key_type)[BIP143]).
 
-While it may seem more intuitive to use a Bitcoin *address* as the identity, the pubkeyhash better fits the definition of identity given in link:./RFC-003-SWAP-Basic.md[RFC003].
-Using a Bitcoin address as the identity would require implementations to do a number of cumbersome validation steps such as verifying that it is a p2pkh or p2wpkh address, extracting the pubkeyhash and validating the network.
+Using `pubkey` will allow migration to Bitcoin mini-scripts, which take public keys, and not pubkeyhashes, as input.
```
Comment (luckysori) [inline RFC-005-SWAP-Basic-Bitcoin.adoc]: :wrench: Seems to be one word.

```suggestion
Using `pubkey` will allow migration to Bitcoin miniscripts, which take public keys, and not pubkeyhashes, as input.
```

Comment (bonomat) [reply RFC-005-SWAP-Basic-Bitcoin.adoc]: Slightly updated your proposal :)

File: RFC-005-SWAP-Basic-Bitcoin.adoc [hunk 2]
```diff
@@ -63,10 +64,10 @@ The parameters for the Bitcoin HTLC follow link:./RFC-003-SWAP-Basic.md#hash-tim
 |secret_hash
 |The hash of the `secret` (32 bytes for SHA-256)
 
-|redeem_identity
+|redeem_pubkeyhash
```
Comment (bonomat) [inline RFC-005-SWAP-Basic-Bitcoin.adoc]: I think we should stick here to `redeem_identity`. Even though the types are different to Ethereum, we still call it the same: 

https://github.com/comit-network/RFCs/blob/master/RFC-007-SWAP-Basic-Ether.md#contract

and 

https://github.com/comit-network/RFCs/blob/master/RFC-007-SWAP-Basic-Ether.md#parameters


```suggestion
| redeem_identity
```

Comment (bonomat) [reply RFC-005-SWAP-Basic-Bitcoin.adoc]: Well.. if looking closely, in the Ethereum RFC we use both: `redeem_identity` and `redeem_address`... so don't mind me :D

**Pull request merged**